backend/src/main/java/com/redhat/cloud/notifications/routers/internal/TemplateResource.java

```diff
@@ -0,0 +1,267 @@
+        return templateRepository.findAggregationEmailTemplatesByApplication(appId);
+    }
+
+    @GET
+    @Path("/email/aggregation/{templateId}")
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public AggregationEmailTemplate getAggregationemailTemplate(@RestPath UUID templateId) {
+        return templateRepository.findAggregationEmailTemplateById(templateId);
+    }
+
+    @PUT
+    @Path("/email/aggregation/{templateId}")
+    @Consumes(APPLICATION_JSON)
+    @Produces(TEXT_PLAIN)
+    @Transactional
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public Response updateAggregationEmailTemplate(@RestPath UUID templateId, @NotNull @Valid AggregationEmailTemplate template) {
+        boolean updated = templateRepository.updateAggregationEmailTemplate(templateId, template);
+        if (updated) {
+            return Response.ok().build();
+        } else {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+    }
+
+    @PUT
+    @Path("/email/aggregation/{templateId}/enable")
+    @Transactional
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public boolean enableAggregationEmailTemplate(@RestPath UUID templateId) {
+        return templateRepository.enableOrDisableAggregationEmailTemplate(templateId, true);
+    }
+
+    @PUT
+    @Path("/email/aggregation/{templateId}/disable")
+    @Transactional
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public boolean disableAggregationEmailTemplate(@RestPath UUID templateId) {
+        return templateRepository.enableOrDisableAggregationEmailTemplate(templateId, false);
+    }
+
+    @DELETE
+    @Path("/email/aggregation/{templateId}")
+    @Transactional
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public boolean deleteAggregationEmailTemplate(@RestPath UUID templateId) {
+        return templateRepository.deleteAggregationEmailTemplate(templateId);
+    }
+
+    @POST
+    @Path("/email/render")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", content = {
+                    @Content(schema = @Schema(title = "RenderEmailTemplateResponseSuccess", implementation = RenderEmailTemplateResponse.Success.class))
+            }),
+            @APIResponse(responseCode = "400", content = {
+                    @Content(schema = @Schema(title = "RenderEmailTemplateResponseError", implementation = RenderEmailTemplateResponse.Error.class))
+            })
+    })
+    @RolesAllowed(RBAC_INTERNAL_USER)
+    public Response renderEmailTemplate(@NotNull @Valid RenderEmailTemplateRequest renderEmailTemplateRequest) {
+        try {
+            return templateEngineClient.render(renderEmailTemplateRequest);
+        } catch (BadRequestException e) {
+            // The following line is required to forward the HTTP 400 error message.
+            return Response.status(BAD_REQUEST).entity(e.getMessage()).build();
```

Comment:
why is this required? I though the BadRequest exception was mapped correctly to the http error.

Reply:
This is old code (already on `master`) that fixes a `quarkus-rest-client-reactive` oddity. It shouldn't be necessary, but it is...

I haven't checked the behavior of that extension since we bumped Quarkus. Maybe they fixed that thing.